graviti/dataframe/frame.py

```diff
@@ -60,38 +57,45 @@ class DataFrame:
 
     """
 
-    _T = TypeVar("_T", bound="DataFrame")
     _columns: Dict[str, Union["DataFrame", ColumnSeries]]
     _column_names: List[str]
     _index: ColumnSeries
+    schema: record
 
-    def __init__(
-        self,
-        data: Union[Sequence[Sequence[Any]], Dict[str, Any], "DataFrame", "LazyLists", None] = None,
-        schema: Any = None,
-        columns: Optional[Iterable[str]] = None,
-    ) -> None:
-        if data is None:
-            data = {}  # type: ignore[assignment]
-            # https://github.com/python/mypy/issues/6463
-        if schema is not None:
-            # TODO: missing schema processing
-            pass
-        if columns is not None:
-            # TODO: missing columns processing
-            pass
-
-        self._columns = {}
-        self._column_names = []
-        if isinstance(data, dict):
-            for key, value in data.items():
-                self._columns[key] = (
-                    DataFrame(value) if isinstance(value, dict) else ColumnSeries(value, name=key)
-                )
-                self._column_names.append(key)
-            self._index = ColumnSeries(list(range(self.__len__())))
-        else:
-            raise ValueError("DataFrame only supports generating from dictionary now")
+    def __new__(
+        cls: "Type[DataFrame]",
+        data: Union[Iterable[NestedDict[str, Any]], "DataFrame", None] = None,
+        schema: Optional[record] = None,
+    ) -> "DataFrame":
+        """Two-dimensional, size-mutable, potentially heterogeneous tabular data.
+
+        Arguments:
+            data: The data that needs to be stored in DataFrame.
+            schema: The schema of the DataFrame. If None, will be inferred from `data`.
+
+        Raises:
+            NotImplementedError: When needed infered schema from the data.
+            ValueError: When the given data is not iterable.
+
+        Returns:
+            The created :class:`~graviti.dataframe.DataFrame` object.
+
+        """
+        if schema is None:
+            if data is None:
+                obj: "DataFrame" = object.__new__(cls)
+                obj._columns = {}
+                obj._column_names = []
+                obj._index = ColumnSeries([])
+                obj.schema = record([])
+                return obj
+
+            raise NotImplementedError("Inferred schema from `data` is not support now.")
+
+        if not isinstance(data, DataFrame) and isinstance(data, Iterable):
+            return DataFrame.from_pyarrow(pa.array(data), schema)
```

Comment:
you should use the schema type to generate the array.
`pa.array(data, schema.to_pyarrow())`